### PR TITLE
Fix SAT NPE and errors

### PR DESCRIPTION
**Pull request by wborn**
The static code analysis issue seems to be caused by the usage of variables using the `var` type in
ItemToStorePointCreatorTest.java , InfluxDBStateConvertUtilsTest.java . When you replace these `var` variables with the actual type the build works. While I was at it I also fixed some other build errors. :wink:

### CODE REVIEW

File: bundles/org.openhab.persistence.influxdb2/src/main/java/org/openhab/persistence/influxdb2/internal/InfluxDB2PersistenceService.java [hunk 1]
```diff
@@ -208,27 +225,31 @@ public void store(Item item, @Nullable String alias) {
...
-        if (logger.isTraceEnabled())
+        if (logger.isTraceEnabled()) {
             logger.trace("MetadataRegistry has been set");
+        }
     }
 
     protected void unsetMetadataRegistry(MetadataRegistry metadataRegistry) {
         this.metadataRegistry = null;
-        if (logger.isTraceEnabled())
+        if (logger.isTraceEnabled()) {
```
Comment (wborn) [inline bundles/org.openhab.persistence.influxdb2/src/main/java/org/openhab/persistence/influxdb2/internal/InfluxDB2PersistenceService.java:250]: You can remove all these if statements whenever the logging doesn't do any expensive operations (like string concatenations). That makes the code more readable and compact.

There's already a check in the logger implementation to only log and substitute logging parameters into strings when the log level is sufficiently high.

File: bundles/org.openhab.persistence.influxdb2/src/test/java/org/openhab/persistence/influxdb2/internal/ItemToStorePointCreatorTest.java [hunk 2]
```diff
@@ -151,18 +169,18 @@ public static String itemName(String line) {
     }
 
     private static Map<String, String> parseFieldValueList(String fieldValueList) {
-        var fields = fieldValueList.split(",", -1);
-        var result = new HashMap<String, String>();
+        String[] fields = fieldValueList.split(",", -1);
+        Map<String, String> result = new HashMap<String, String>();
```
Comment (wborn) [inline bundles/org.openhab.persistence.influxdb2/src/test/java/org/openhab/persistence/influxdb2/internal/ItemToStorePointCreatorTest.java:173]: You can write less code using the diamond operator:
 
```suggestion
        Map<String, String> result = new HashMap<>();
```

**Review by lujop — APPROVED** [review]
Thanks for the corrections.
I will check at the formatting, although I use IntelliJ, I use the Eclipse formatted plugin with ESH.xml rules files.

But looking at your corrections, it seems that it's not exactly the same

**Comment by wborn** [comment]
Great! I already thought you were using another IDE. :wink: Most of the changes like imports and curly braces were automatically done by Eclipse after saving. I only had to manually correct some license headers and some `@author` tags. But it wasn't that much work and it's always nice to see the build succeed. :-)

**Pull request merged**